pkg/server/singleprocess/poll_application.go

```diff
@@ -240,7 +240,12 @@ const (
 )
 
 // appStatusPollSingletonId generates an application status polling job singleton ID
-// for the given app and operation type.
-func appStatusPollSingletonId(appName string, operationType appStatusPollOperationType) string {
-	return fmt.Sprintf("app-status-poll/%s/%s", appName, operationType)
+// for the given workspace, project, app and operation type.
+func appStatusPollSingletonId(
```

Comment:
At this point I kind of wish we had some had a structured message here - this is a whole lot of info to encode positionally into a string. Understand if that's a bigger harder refactor for now though.

Reply:
Yeah, the main idea here for why I originally had a structured singleton ID is that the same poll handler per app shouldn't fire off more than 1 job at a time for status reporting. I think this is about as much info as we'd want in a singleton ID to keep poll jobs unique. Otherwise there's a bug here in Waypoint server where any app in separate projects would block on a separate job running given they'd have the same singleton id.

Reply:
Added a note! The reasoning here might of been only in my head, so calling it out explicitly hopefully will clear things up for why we're doing this.